### PR TITLE
Deprecate `Primer::Truncate` in favor of `Primer::Beta::Truncate`

### PR DESCRIPTION
**Pull request by hrs**
### Description

This deprecates `Primer::Truncate` in favor of `Primer::Beta::Truncate` and provides a migration guide.

It also fixes some trivial typos in a couple other migration guides.

### Integration

> Does this change require any updates to code in production?

Nope! `Primer::Truncate` doesn't seem to be used any more in dotcom.

### Merge checklist

- [ ] Added/updated tests
- [X] Added/updated documentation
- [ ] Added/updated previews

### CODE REVIEW

**Review by keithamus — APPROVED** [review]
LGTM, thanks!

File: docs/content/guides/primer_truncate.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+| `tag`                   | `tag`                       | Previously defaulted to `:div`, now `:span`.                             |
+| `inline`                | n/a                         | Removed in `Primer::Beta::Truncate`.                                     |
+| n/a                     | `priority`                  | If `true`, the text will be given priority (by increasing `flex-basis`). |
+
+The remaining arguments for `Truncate` can be found in the documentation for
+that component.
+
+Please see the following for complete descriptions and examples.
+
+* [Deprecated `Primer::Truncate`](https://primer.style/view-components/components/truncate)
```
Comment (jonrohan) [inline docs/content/guides/primer_truncate.md:61]: Can you add a link to the old page under the deprecated section 
image

File: docs/content/guides/primer_truncate.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+<%= render(Primer::Beta::Truncate.new(tag: :div)) { "Some more very very long text" } %>
+```
+
+## Arguments
+
+The following arguments for the component initializer have changed between the deprecated and newer versions
+of the `Truncate` component.
+
+| From `Primer::Truncate` | To `Primer::Beta::Truncate` | Notes                                                                    |
+|-------------------------+-----------------------------+--------------------------------------------------------------------------|
```
Comment (jonrohan) [inline docs/content/guides/primer_truncate.md:51]: The markdown parser might be angry about the `+` here?

https://primer-075f5335f7-26611710.drafts.github.io/guides/primer_truncate

image

File: docs/content/guides/primer_truncate.md (hunk 1 above)
Comment (hrs) [reply docs/content/guides/primer_truncate.md:61]: Sure thing, good idea!

File: docs/content/guides/primer_truncate.md (hunk 2 above)
Comment (hrs) [reply docs/content/guides/primer_truncate.md]: Oof, I've been writing in Org-mode too long. Fixed!

**Review by jonrohan — APPROVED** [review]
Looks great 👍🏻

**Comment by camertron** [comment]
FYI there are ~110 usages of `Primer::Truncate` in dotcom 😅

**Pull request merged**